code/modules/mob/living/simple_animal/hostile/mining_mobs/brimdemon.dm

```diff
@@ -0,0 +1,189 @@
+#define BRIMBEAM_RANGE 10
+
+/mob/living/simple_animal/hostile/asteroid/brimdemon
+	name = "brimdemon"
+	desc = "A misshapen demon with big, red eyes and a hinged mouth. Not much is known about the creatures \
+		due to their response to any unexpected stimulus being \"brimbeam\", a deadly blood-laser barrage."
+	icon = 'icons/mob/brimdemon.dmi'
+	icon_state = "brimdemon"
+	icon_living = "brimdemon"
+	icon_dead = "brimdemon_dead"
+	mob_biotypes = MOB_ORGANIC|MOB_BEAST
+	speak_emote = list("cackles")
+	emote_taunt = list("screeches")
+	emote_hear = list("cackles","screeches")
+	combat_mode = TRUE
+	stat_attack = HARD_CRIT
+	ranged_cooldown_time = 5 SECONDS
+	vision_range = 9
+	retreat_distance = 2
+	speed = 5
+	move_to_delay = 5
+	maxHealth = 200
+	health = 200
+	obj_damage = 15
+	melee_damage_lower = 7.5
+	melee_damage_upper = 7.5
+	rapid_melee = 2 // every second attack
+	attack_verb_continuous = "bites"
+	attack_verb_simple = "bite"
+	attack_sound = 'sound/weapons/bite.ogg'
+	attack_vis_effect = ATTACK_EFFECT_BITE
+	butcher_results = list(/obj/item/food/meat/slab = 2)
+	loot = list()
+	robust_searching = TRUE
+	footstep_type = FOOTSTEP_MOB_CLAW
+	deathmessage = "wails as infernal energy escapes from its wounds, leaving it an empty husk."
+	deathsound = 'sound/magic/demon_dies.ogg'
+	light_color = LIGHT_COLOR_BLOOD_MAGIC
+	light_power = 5
+	light_range = 1.4
+	crusher_loot = /obj/item/crusher_trophy/brimdemon_fang
+	/// Are we charging/firing? If yes stops our movement.
+	var/firing = FALSE
+	/// A list of all the beam parts.
+	var/list/beamparts = list()
+
+/mob/living/simple_animal/hostile/asteroid/brimdemon/Login()
+	ranged = TRUE
+	return ..()
+
+/mob/living/simple_animal/hostile/asteroid/brimdemon/Logout()
+	ranged = FALSE
+	return ..()
+
+/mob/living/simple_animal/hostile/asteroid/brimdemon/death()
+	firing = FALSE
+	cut_overlay("brimdemon_telegraph_dir")
+	move_resist = initial(move_resist)
+	return ..()
+
+/mob/living/simple_animal/hostile/asteroid/brimdemon/Goto(target, delay, minimum_distance)
+	if(firing)
+		return FALSE
+	return ..()
+
+/mob/living/simple_animal/hostile/asteroid/brimdemon/MoveToTarget(list/possible_targets)
+	if(firing)
+		return FALSE
+	return ..()
+
+/mob/living/simple_animal/hostile/asteroid/brimdemon/AttackingTarget(atom/attacked_target)
+	if(firing)
+		return FALSE
+	return ..()
+
+/mob/living/simple_animal/hostile/asteroid/brimdemon/Move(atom/newloc, dir, step_x , step_y)
+	if(firing)
+		return FALSE
+	return ..()
+
+/mob/living/simple_animal/hostile/asteroid/brimdemon/OpenFire()
+	if(firing)
+		to_chat(src, "<span class='warning'>You are already firing!</span>")
```

Comment:
```suggestion
		to_chat(src, span_warning("You are already firing!"))
```
same for the other spans

Reply:
its april 2021 code ill go ahead and make it balloon alerts